src/main/java/Parser.java

```diff
@@ -0,0 +1,127 @@
+                        break;
+                    case "list":
+                        ui.listTasks(taskList);
+                        break;
+                    case "bye":
+                        ui.exitDuke();
+                        loop = false;
+                        break;
+                    default:
+                        throw new InvalidCommandException("");
+                    }
+            } catch (InvalidCommandException | InvalidTodoException
+                    | InvalidDeadlineException | InvalidEventException ex) {
+                System.out.println(ex);
+            }
+        } while (loop);
+    }
+
+    public static String extractCommandType(String input) throws InvalidCommandException{
+        String[] inputSplit = input.split(" ", 2);
+        boolean isValidCommand = checkValidCommand(inputSplit[0]);
+
+        if (!isValidCommand) {
+            throw new InvalidCommandException("");
+        }
+
+        return inputSplit[0];
+    }
+
+    public static String prepareTodo(String[] input) throws InvalidTodoException {
+        if (!checkValidTodo(input)) {
+            throw new InvalidTodoException("");
+        } else {
+            return input[1];
+        }
+    }
+
+    public static String[] prepareDeadline(String[] input) throws InvalidDeadlineException {
+        String[] fieldDetails = input[1].split("/", 2);
+
+        if (!checkValidDeadline(fieldDetails)) {
+            throw new InvalidDeadlineException("");
+        } else {
+            return fieldDetails;
+        }
+    }
+
+    public static String[] prepareEvent(String[] input) throws InvalidEventException {
+        String[] fieldDetails = input[1].split("/", 2);
+
+        if (!checkValidEvent(fieldDetails)) {
+            throw new InvalidEventException("");
+        } else {
+            return fieldDetails;
+        }
+    }
+
+    public static int prepareDoneDelete(String input) {
+        return Integer.parseInt(input);
+    }
+
+    public static boolean checkValidCommand(String type) {
+        if (type.equals("todo")
+                || type.equals("deadline")
+                || type.equals("event")
+                || type.equals("list")
+                || type.equals("done")
+                || type.equals("delete")
+                || type.equals("bye")) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public static boolean checkValidTodo(String[] input) {
+        if (input.length == 2) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public static boolean checkValidDeadline(String[] input) {
+        if (input.length == 3) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public static boolean checkValidEvent(String[] input) {
+        if (input.length == 3) {
+            return true;
+        } else {
+            return false;
+        }
+    }
```

Comment:
I love the abstraction here, and it would be even better if these methods were made private.